### PR TITLE
Add helm chart for KServe

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
This PR adds the helm chart for installing kserve and modelmesh

## Install KServe
`helm install kserve manifests/charts -n kserve`

## Upgrade KServe
 `kubectl apply -f manifests/charts/crds`
`helm upgrade kserve manifests/charts -n kserve`

## Uninstall KServe
`helm uninstall kserve`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: manifests/charts/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        command:
+        - /manager
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: ETCD_SECRET_NAME
+          value: model-serving-etcd
+        image: {{ .Values.kserve.modelmesh-controller.image }}
```
Comment (pvaneck) [inline manifests/charts/templates/deployment.yaml]:
```suggestion
        image: {{ .Values.kserve.modelmesh.controller.image }}
```

File: manifests/charts/templates/rolebinding.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  namespace: kserve
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  labels:
+    app.kubernetes.io/instance: modelmesh-controller
+    app.kubernetes.io/managed-by: modelmesh-controller
+    app.kubernetes.io/name: modelmesh-controller
+  name: controller-leader-election-rolebinding
```
Comment (pvaneck) [inline manifests/charts/templates/rolebinding.yaml]:
```suggestion
  name: modelmesh-controller-leader-election-rolebinding
```

File: manifests/charts/templates/rolebinding.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: controller-leader-election-role
```
Comment (pvaneck) [inline manifests/charts/templates/rolebinding.yaml]:
```suggestion
  name: modelmesh-controller-leader-election-role
```

File: manifests/charts/templates/rolebinding.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+metadata:
+  labels:
+    app.kubernetes.io/instance: modelmesh-controller
+    app.kubernetes.io/managed-by: modelmesh-controller
+    app.kubernetes.io/name: modelmesh-controller
+  name: modelmesh-controller-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: controller-role
```
Comment (pvaneck) [inline manifests/charts/templates/rolebinding.yaml]:
```suggestion
  name: modelmesh-controller-role
```

File: manifests/charts/values.yaml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        cpu: 100m
+        memory: 300Mi
+  modelmesh:
+    controller:
+      image: kserve/modelmesh-controller:v0.7.0
+    config:
+      modelmeshImage: kserve/modelmesh
+      modelmeshImageTag: v0.7.0
+      modelmeshRuntimeAdapterImage: kserve/modelmesh-runtime-adapter
+      modelmeshRuntimeAdapterTag: v0.7.0
```
Comment (pvaneck) [inline manifests/charts/values.yaml]:
```suggestion
      modelmeshRuntimeAdapterImageTag: v0.7.0
      restProxyImage: kserve/rest-proxy
      restProxyImageTag: v0.1.1
```

**Review by pvaneck — CHANGES_REQUESTED** [review]
Thanks, @yuzisun. For the modelmesh related yaml, I had to make a few changes to get things to work which I added inline. After that, the deployment using helm worked well, and I as able to deploy a predictor (after configuring etcd).

**Comment by yuzisun** [comment]
> Thanks, @yuzisun. For the modelmesh related yaml, I had to make a few changes to get things to work which I added inline. After that, the deployment using helm worked well, and I as able to deploy a predictor (after configuring etcd).

Thanks @pvaneck ! Forgot to run another test after adding MM template variables, updated the PR now.

**Review by pvaneck — COMMENTED** [review]
/lgtm

**Pull request merged**